### ISSUE TITLE
Syntax error in regexp pattern

### ISSUE THREAD

**Issue by adamemerson** [matburt/mobileorg-android#472]
I am getting Syntax Error in Regexp Pattern and, after I sync, while it shows the file correctly, it's empty. There's nothing in it.

Also, I can't even read the entire error message. It's truncated and there doesn't seem to be a way to find the full text.

I am using emacs 24.5.1
org-20151005
and
MobileOrg 0.9.13

This is a base64 encoding of a .tar.bz2 of the staging area. I tried to attach it but github is being a stupid twit.

QlpoOTFBWSZTWbFS6L8ADAZ/hP2wAIBKf//3P///uv//3/AACAAIYAd31CQpSAIANYQAoGppQAAA
aGgABoANAB6gAAABJITIACaNRqeieqbU9J+lDQGCaaNkJpmkD1MhkOaNGhphANMCaaAMhoYgDRiN
DBGQA5o0aGmEA0wJpoAyGhiANGI0MEZADmjRoaYQDTAmmgDIaGIA0YjQwRkACpJCniTI0MgE0yU9
kU2p6mZNNTCD1APEh5R5NR6T0C7PUFWzRG74waYTOZJAw04kIIdqLbu3LUoCwJR1139nXouzxjJu
xKSmmpCdIuCXqoKzLpv+MH/K4awev1gQE8XwboahzkQYWmLx8PcwWqsuqlouMWJTEXDfKK/Lzf0z
lZ+a5i1qUnKuWU/zu4tF9p07ZypvAY1HClSImMmwvUpYLFSFlCxQLiRyjp/X7eimA0R8v9rlR0/V
LN5Jz3/HGnbRVFHAZSTi+v3V0VkzAKSSPB4ctJgOr5xUTLuMcp4DQ0OByG898tzBQn000TOZgmNJ
Q3arsAZg6feTj6Z6QuYp32vC8+GpPv+iqqOOKhUWtVVVUYLzT1HI0uzS5WOPJVu7rR1dnh8sqWau
7gxx8fPA66y1OVtkNNQpRXhpZFKUqkqhQpJ6ZNXm3PAq0c30zQaLSxa0qp4cezrTKXRdYta0Oi1o
6OzSbeh6IokhkdiIo2EEclFVKIR1iNoejr5c27zHzpVrGWeyNW9vb0cPSi++/LLG4+PPEcUkW9RV
3bJFwu7ZIwYb8WWpVKpVIqLXrbkkd5FNCOKwWCRdcXSciAkdfX7fabxB2RQuA+gAqKIuJnW/gnLg
6tfAmE11IrCIZpj5LsY4LT1cWTJfFau6rXt5ZyR85cqMui6PO1l8Z570QE/DITkBWhppnCQ+meSI
AprcMO6iwtmZQ40V1JMieOMiCXdRmjJYeft8TDDDDCnEnOQ/qZCCtliiqTs3JBe/PPZxwnE91U3V
QGt+b0sEZIreKpKLAMjJVV9E7bb57GSrwKsrkgmal7QtbWiaMLlGNPYDRNtVyMrET4TsjERZx3Vg
voGAvWIs6hRlATqrJ9VaLwAfLCfZ9M1+4mtL0leh4p9iMn48iKCRXFGlxYURLOJGxEbb0a0ObNEI
ApX0mDIlYiSHDTVOQ5EUYE9DIMkOozRW6ELZkWvutNDQoKoo14USRgjOe2N1qH3VFMEVvxADVojb
W9p2+yHvRVeIsn2BdXnVPTejxtkkZczXqtBQbrQpT1sU1jnKFKFiYaa4++MAp9z7xwmkZlCj+AoY
8SHotPZ5Rc8l812st3Ufjdvpdij/SaEowvxosVR6g2RLSWWcbKSOI6A8zYmpHSjVI+40/Clw1cIe
0jw5viOORqHuGsLLjLLJ/c3JgcAbp60v4lw3z6mscyOSSwUziaJJUN9HCMN1M0mDP4M0YYo3MHSw
HJVI0o1KRsr2mCbjJGabJJRQ2o3byW27nM4UcO4UVIvqXKYQMrZ5cYuhsVPcpL0cTWXb1J4IRUIw
hFoDSvQpnUXADLbMAN5Aw2rEAwJQBBASiCDNrBzEciqrdeBay8ptGxNfpe+ps9H2tWf7eY+2NzJr
jazj2xHVIbvo+QOkPIOr5mDRJg9SVocBW/24ngJbBOSR/QvxRlftRmMj8tSPLfkjNFYGYH4ICw3i
5kcr0Xszi8th0oqqe7nPsEuJLZAsxYdiHDfTJ72vfJDmD1844YY/wqHmdeqaJ/yQu9REkUCtRj7x
LpI7Y4+rlujmy5IRiG1H5e71tGixoC2+z+mGhLMiHixSUNba5+6Qc+whzXsUcf6o2R3o44rijGRP
DHnhG0PHiNshT8CpOEl+9ofZakH6KJ2yR12LENDm+XihHQwywJ6GiJxMw08EtES3ec9M+GTnhGql
nudCOqSkbtEk1a87sKUivrVuorJOXQ0QxYN1rUSyPwap6ngOVck4WzgpXYjwI8/NJHyIbz66lKR0
h1GxKKVPxkSeJFxC6RNKLciUkZ+Plkk5ApKkPYqIP5bWWC8hsttVfyVVUuuSw87uF1ODFvDh94iI
n+Tm17fH3Oi7BLnpBB9wjelCkDtCHX8DlwR2d9hgk5GLB8/E7ZuSatpOp+cgyfR5KSvmlkfneQ4w
6ZIqB5dfXSSpBq1UnTfIKQn/xdyRThQkLFS6L8A=

**Comment by stvharman** [comment]
I have the same problem. I cannot see the content.

**Comment by jersey99** [comment]
If it is what I think it is, it is a bug introduced in the latest org-mobile code. This worked for me.
https://lists.gnu.org/archive/html/emacs-orgmode/2015-09/msg00159.html

**Comment by stvharman** [comment]
Yes, this works. thanks

**Comment by stvharman** [comment]
When I come to the end of the function org-mobile-create-index-file and then evaluate that function, the change takes effect. Then M-x org-mobile-push successfully stages the necessary files. However, when emacs re-starts, I need to do re-evaluate this function again. How can I make it automatic? I tried to byte-compile-file the entire org-mobile.el file to create org-mobile.elc after the modification and after deleting the old compiled file (org-mobile.elc). However, it didn't work. Any help would be appreciated. Thanks. Steve

**Comment by hugoroy** [comment]
I also have this problem. It looks like MobileOrg is in need of an update...

**Comment by hugoroy** [comment]
> If it is what I think it is, it is a bug introduced in the latest org-mobile code. This worked for me.
> https://lists.gnu.org/archive/html/emacs-orgmode/2015-09/msg00159.html

It does not work for me. Did I miss something?

**Comment by pedros007** [comment]
I experienced the same thing with

* emacs-24.5.1 on my Mac installed via Homebrew, 
* org-20151005
* org-mobile-sync-20131118.1116 installed via ELPA

I tried applying the diff at https://lists.gnu.org/archive/html/emacs-orgmode/2015-09/msg00159.html without luck.  Eventually I just installed [Orgzly](http://www.orgzly.com/), which also has [DropBox sync support](http://www.orgzly.com/help/#sec-1-1-1) and seems to work fine.  Orgzly is [currently closed source but it "is now going in that direction"](http://www.orgzly.com/help/#Source).

**Comment by stvharman** [comment]
Folks, who read this or similar messages elsewhere. Orgzly is not the solution if it is closed source. Free software is something very dear to our hearts, important for the free society, and development of knowledge and science. So, please do not use Orgzly unless they accept to release their code under GPL.

**Comment by hugoroy** [comment]
Any update on this? 

I have to use my firm's iPhone in order to access org mode on mobile. What a shame :-(

**Comment by simendsjo** [comment]
The small patch linked by @jersey99 worked for me. Took me a long time to find this issue though. This has obviously been a problem for at least 8 months. What's wrong with the proposed patch?

**Comment by hugoroy** [comment]
Maybe time to fork this if the former maintainer is not taking care of it anymore.

**Comment by smartass101** [comment]
As a workaround I set
```org-mode
#+TODO: CONSIDER(c) TODO(t) MAYBE(m) | DONE(d)
```
in one of my agenda files. Seems to fool the parser enough for it to process it.

**Comment by eedahl** [comment]
I'm having this issue too.

**Comment by simendsjo** [comment]
@smartass101 This workaround didn't work for me.

**Comment by clinty** [comment]
@simendsjo I needed to put it in every single file

**Comment by jorgemorais** [comment]
> As a workaround I set
> 
> #+TODO: CONSIDER(c) TODO(t) MAYBE(m) | DONE(d)
> in one of my agenda files.  Seems to fool the parser enough for it to process it.  The trick seems to be an explicit | symbol somewhere which the parses expects.
> [...]
> @simendsjo I needed to put it in every single file

This workaround worked for me too, except that, since I don’t use per-file todo keywords, what I did was customize the option org-todo-keywords.  This option was:
`(org-todo-keywords`
`   (quote`
`    ((sequence "TODO(t@)" "FBCK_(F@)" "CNTN_(O@)" "FBCK(f@)" "CNTN(o@)" "DONE(d@)")`
`     (sequence "CNCD(c@)")`
`     (sequence "DGTD(e@)")`
`     (sequence "FIX_COMM(i@)"))))`

The workaround consisted in customizing it to:
`(org-todo-keywords`
`   (quote`
`    ((sequence "TODO(t@)" "FBCK_(F@)" "CNTN_(O@)" "FBCK(f@)" "CNTN(o@)" "|" "DONE(d@)")`
`     (sequence "|" "CNCD(c@)")`
`     (sequence "|" "DGTD(e@)")`
`     (sequence "|" "FIX_COMM(i@)"))))`

Note that in every sequence of todo keywords, the DONE keywords are now separated with a vertical bar.

After that, I called org-mobile-push and then, in Android MobileOrg, I cleared the database and synchronized again.  Now the problem is fixed for me.
